### PR TITLE
Fix Japanese Internal Names

### PR DESCRIPTION
**Pull request by Frank-74**
japnames

### CODE REVIEW

File: Source/Common/std string.cpp [hunk 1]
```diff
@@ -241,3 +241,31 @@ std::wstring stdstr::ToUTF16 ( bool * bSuccess)
 	}
 	return res;
 }
+
+std::wstring stdstr::ToUTF16(uint32_t CodePage, bool * bSuccess)
```
Comment (project64) [inline Source/Common/std string.cpp]: I like the change to use the code page to get the correct internal name, needs to also be applied to the rom information dialog.

But I dislike duplicating ToUTF16 can we not just add the CodePage param to the current version with a defualt of CP_UTF8?

**Comment by Frank-74** [comment]
Ok, I'll apply it to the Rom Information.  And add the CodePage param to the existing version as suggested.  I just copy/pasted @Nekokabu 's method to get it working.

It's 9am here, been up all night, need sleep.  Will make the changes tomorrow.  Thanks.

**Comment by Frank-74** [comment]
@project64 I cannot get it working.  If I add the CodePage param to the existing version, I get errors.
> : error C2660: 'stdstr::ToUTF16' : function does not take 0 arguments

And using W instead of A, has broken reading of non english filenames.  I think https://github.com/project64/project64/commit/5b7020453e184ccc2af04b5cd5b0d9b1d95aa894 needs to be reverted.

**Comment by project64** [comment]
ok I made some changes to stdstr (https://github.com/project64/project64/commit/942e07569afb4416028405d141a97e387add52c5) to add codepages, and (https://github.com/project64/project64/commit/655a73e1face690986901a1f9f9dd279e3f308f7) to use code page 932

**Comment by Frank-74** [comment]
@project64 Just built the current master.

Internal Name is ok, but all japanese filenames don't show up in the ROM browser at all.

I used google translate to rename a J game to Japanese language.
Air Boarder 64 (J) [!] translates to > エアボーダー64（J）[！]

**Comment by Frank-74** [comment]
Also, changing just this String to use code page 932, fixes the Internal Names also.
https://github.com/project64/project64/blob/master/Source/Project64/User%20Interface/Rom%20Browser%20Class.cpp#L1243

So https://github.com/project64/project64/commit/655a73e1face690986901a1f9f9dd279e3f308f7 wouldn't be needed.

Probably fixes the filenames as well, if they would show up.

**Comment by project64** [comment]
the:
AddRomToList(stdstr((std::string &)SearchPath).ToUTF16().c_str(), stdstr(lpLastRom).ToUTF16().c_str());

might just need a different code page in the name conversion .. or I need to do the searching for names in the W version

**Comment by Frank-74** [comment]
From > https://support.microsoft.com/en-us/kb/108450

```
CP_ACP instructs the API to use the currently set default Windows ANSI codepage. CP_OEMCP instructs the API to use the currently set default OEM codepage.

If Win32 ANSI APIs are used to get filenames from a Windows NT system, use CP_ACP when converting the string. Windows NT retrieves the name from the physical device and translates the OEM name into Unicode. The Unicode name is translated into ANSI if an ANSI API is called, then it can be translated back into Unicode with MultiByteToWideChar().

**Comment by project64** [comment]
ok does using CP_ACP or  CP_OEMCP allow the files to be listed ?

**Comment by Frank-74** [comment]
> ok does using CP_ACP or CP_OEMCP allow the files to be listed ?

@project64 It doesn't allow the files to be listed here.  Tried also codepage 932.

I think @Nekokabu needs to test if it works for him.  It might need a Windows with Japanese as default.

Can't you just rename a file using google translate to test it?
https://translate.google.co.uk/?ie=UTF-8&hl=en&client=tw-ob#en/ja/

**Comment by Frank-74** [comment]
I don't know if this is useful, but I found this comment in a post.

> . A filename selected (e.g. from a file
open dialog box) that include Chinese characters will fail if a
boost::filesystem::path object is constructed from the const char*
ANSI path. If we just pipe the const char* directly into std::ifstream
then it works ok.

**Comment by project64** [comment]
I will do some testing tonight, I might have to keep the path as ansi and just translate for display or look at converting path class to work as W function ..

the first one might be the easiest to do....

**Comment by project64** [comment]
I changed the file handling back to ansi so it should be back to what it was originally. I tried making a file with jap file name and the ansi file name returned is useless. So no idea how it would work originally. I might need to change file handling to wide char through the whole application

**Comment by Frank-74** [comment]
Just had a little experiment using File > Open ROM.

Japanese named files return 'Failed to find filename in 7z file'.

Zipping up the same Japanese file, but name the 7z with English, and it opens ok.

**Comment by project64** [comment]
I do not see how non ansi files ever worked, this is as close as I can get with the ROM browser I believe with out changing all file handling to use utf-16 which I am not inclined to do at this point of time

**Comment by Frank-74** [comment]
@project64 
@Nekokabu found a way in his original PR, he showed it working for him in the last comment i this issue.  https://github.com/project64/project64/issues/731

**Comment by cxd4** [comment]
Codepage 932 doesn't support any of my filenames written in Arabic gibberish.

**Comment by Frank-74** [comment]
@cxd4 Do Arabic named files display in the ROM browser Filename column correctly?

**Comment by cxd4** [comment]
Dunno, I was trolling.

I think English is the ANSI-standard language for flamewars and trolling on the interwebz, so people should learn English or they won't survive.  (Or at least some Latin-based language that allows their N64 file names to be normal ones.)

932 sounds as good to me as about anything else, although some guy I am sure is going to rename all their things to be written in gibberish sign-language characters that would probably just break something no matter how it's done anyway.

**Comment by cxd4** [comment]
I do like seeing the ROM "internal name" fields in the correct language though in the browser, for those Japanese titles that use NOJ's font code page.

**Pull request closed without merging**